### PR TITLE
Select Into support

### LINKED ISSUE #6148: SELECT INTO Support
### Is your feature request related to a problem or challenge?

Add support for queries in the form of
```
SELECT ts
INTO table2
FROM table1
```

### Describe the solution you'd like

SELECT ... INTO and CREATE TABLE ... AS gives the same results AFAIK, like:

```
CREATE TABLE orders_summary AS
SELECT c.customer_id, c.first_name, c.last_name, COUNT(o.order_id) AS total_orders, SUM(o.order_total) AS total_order_value
FROM customers c
JOIN orders o ON c.customer_id = o.customer_id
GROUP BY c.customer_id;
SELECT * FROM orders_summary;
```

```
SELECT c.customer_id, c.first_name, c.last_name, COUNT(o.order_id) AS total_orders, SUM(o.order_total) AS total_order_value
INTO orders_summary
FROM customers c
JOIN orders o ON c.customer_id = o.customer_id
GROUP BY c.customer_id;
SELECT * FROM orders_summary;
```

SELECT INTO can construct the same statement as CREATE TABLE AS.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by berkaysynnada**
# Which issue does this PR close?

Closes #6148.

# Rationale for this change

We can support the queries in the form of

```
SELECT ts
INTO table2
FROM table1
```

# What changes are included in this PR?

`select_to_plan()` function that builds the select plans is extended by one step. At the end of the build, since the `CreateMemoryTable` plans are placed at the top of the other plans, `into` option is checked. If it exists, CreateMemoryTable logical plan is added.

SELECT INTO statement does not copy constraints such as primary key; therefore, primary_key input is given empty to the constructor. SELECT INTO also should give error in case of creating a table having the same name with an existing table, so if_not_exists and or_replace option is given false.

# Are these changes tested?

Yes

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/sql/src/select.rs [hunk 1]
```diff
@@ -221,7 +219,23 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 .collect::<Result<Vec<_>>>()?;
             LogicalPlanBuilder::from(plan)
                 .repartition(Partitioning::DistributeBy(x))?
-                .build()
+                .build()?
+        } else {
+            plan
+        };
+
+        if let Some(select_into) = select.into {
```
Comment (alamb) [inline datafusion/sql/src/select.rs:227]: I verified that this is consistent with postgres:

```sql

postgres=# select * from test;
 column1 | column2
---------+---------
       1 |      10
       2 |      20
       3 |      30
(3 rows)

postgres=# select * into test2 from test;
SELECT 3
postgres=# select * into test2 from test;
ERROR:  relation "test2" already exists
postgres=# select * from test2;
 column1 | column2
---------+---------
       1 |      10
       2 |      20
       3 |      30
(3 rows)
```

File: datafusion/core/tests/sqllogictests/test_files/ddl.slt [hunk 2]
```diff
@@ -296,9 +320,20 @@ SELECT * FROM my_table;
 ----
 1 2 hello
 
+# select_into_with_schema_as_values
+statement ok
+SELECT * INTO new_table FROM my_table
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/ddl.slt:325]: I don't understand how this test is different than the ones above. Is the intent to insert into a table in a schema? Like `SELECT * into my_schema.new_table from my_table`?

File: datafusion/core/tests/sqllogictests/test_files/ddl.slt [hunk 3]
```diff
@@ -476,6 +511,15 @@ SELECT * FROM abc
 1 3 3
 5 5 6
 
+statement ok
+SELECT * INTO fgh FROM (VALUES (1,2,3), (4,5,6));
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/ddl.slt:515]: likewise it isn't clear to me what other case this is testing

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @berkaysynnada 

I had some questions about some of the tests -- once they are cleared up I think this PR is good to go.

File: datafusion/core/tests/sqllogictests/test_files/ddl.slt [hunk 4]
```diff
@@ -28,6 +28,18 @@ select * from users;
 1 2
 2 3
 
+statement ok
+drop table users
+
+statement ok
```
Comment (alamb) [inline datafusion/core/tests/sqllogictests/test_files/ddl.slt:34]: Can you also please add a negative test (aka that calling `select * INTO users ...` will error if `users` already exists?

**Comment by ozankabak** [comment]
@berkaysynnada, seems like we have a minor Cargo.lock issue making CI fail

File: datafusion/core/tests/sqllogictests/test_files/ddl.slt (hunk 2 above)
Comment (alamb) [reply datafusion/core/tests/sqllogictests/test_files/ddl.slt:329]: I think the comments have made the intent clear -- thank you

**Review by alamb — APPROVED** [review]

**Pull request merged**